Common/Rntbd/RntbdTokenStream.cs

```diff
@@ -87,7 +87,7 @@ public void Reset()
         private static int[] GetTokenPositions(RntbdToken[] t)
         {
             Debug.Assert(t != null);
-            TokenPositionMap = new int[t.Select(e => e.GetTokenIdentifier()).Max()];
+            TokenPositionMap = new int[t.Select(e => e.GetTokenIdentifier()).Max() + 1];
```

Comment:
This potentially is a bug on the original code, if the token list has 5 identifiers (0 to 4) this creates an array of 4 (max is 4) items, which when looping through, it generates an OutOfRangeException because the map only contains only 4 (0 to 3) positions.